ui.frontend/src/view/FormFieldBase.js

```diff
@@ -602,7 +608,17 @@ class FormFieldBase extends FormField {
      */
     updateDescription(description) {
         if (this.description) {
-            this.description.querySelector("p").innerHTML = description;
+            if (description) {
+                if (!this.description.querySelector("p")) {
+                    // If the description is updated via rule then it might not have <p> tags
+                    this.description.appendChild(document.createElement('p'));
+                }
+                this.description.querySelector("p").innerHTML = description;
+                const questionMarkDiv = this.getQuestionMarkDiv();
+                 if (questionMarkDiv) {
+                     questionMarkDiv.style.display = "block";
```

Comment:
We should not be updating styles via code, this might override the theme, and might conflict. What is the use-case to override style

Reply:
The requirement is that if long description is not set during authoring then the questionMark div and description div should be present in DOM but hidden. And if by any rule the description is set in runtime, then they should be visible. I guess we could do that by adding 'data-cmp-visible' attribute instead

Reply:
yes, please check this comment on the overall design, https://github.com/adobe/aem-core-forms-components/pull/1274#discussion_r1654523515